From b445e9c68f39e56383c91fafc8f7e50e213a8198 Mon Sep 17 00:00:00 2001
From: Bo Shen <voice.shen@atmel.com>
Date: Wed, 11 Mar 2015 10:32:44 +0800
Subject: [PATCH] Config: remove dependency in the sub category

Remove the dependency in the sub category, it handle in the
first level.

Signed-off-by: Bo Shen <voice.shen@atmel.com>
---
 driver/Config.in.dataflash | 9 ---------
 driver/Config.in.memory    | 8 ++++++--
 driver/Config.in.nandflash | 2 --
 3 files changed, 6 insertions(+), 13 deletions(-)

diff --git a/driver/Config.in.dataflash b/driver/Config.in.dataflash
index b03b5be..81f6d38 100644
--- a/driver/Config.in.dataflash
+++ b/driver/Config.in.dataflash
@@ -1,9 +1,7 @@
 menu  "SPI configuration"
-	depends on CONFIG_DATAFLASH
 
 config	CONFIG_SPI_CLK
 	int "SPI clock speed"
-	depends on CONFIG_DATAFLASH
 	default 33000000
 	help
 	  Which speed (in Hz) should the SPI run at.
@@ -11,7 +9,6 @@ config	CONFIG_SPI_CLK
 config CONFIG_SMALL_DATAFLASH
 	bool "Support < 32 Mbit dataflashes"
 	default	y
-	depends on CONFIG_DATAFLASH
 	help
 	  Deselect this to save some bytes of memory
 	  at the expense of flexibility in selecting memory sizes.
@@ -19,7 +16,6 @@ config CONFIG_SMALL_DATAFLASH
 config CONFIG_DATAFLASH_RECOVERY
 	bool "Support Dataflash recovery by pressing a button"
 	default y
-	depends on CONFIG_DATAFLASH
 	depends on ALLOW_DATAFLASH_RECOVERY
 	help
 	  Let bootstrap disassemble the first 7 double words
@@ -33,29 +29,24 @@ config ALLOW_DATAFLASH_RECOVERY
 # ------- SPI boot source -----------------------------------------------------
 choice
 	prompt "Chip Select"
-	depends on CONFIG_DATAFLASH
 	default CONFIG_SPI_CS0
 	help
 	  Determine which SPI chip select is to be used
 
 config	CONFIG_SPI_BOOT_CS0
 	bool	"Boot from chip select 0"
-	depends on CONFIG_DATAFLASH
 	depends on ALLOW_BOOT_FROM_DATAFLASH_CS0
 
 config	CONFIG_SPI_BOOT_CS1
 	bool	"Boot from chip select 1"
-	depends on CONFIG_DATAFLASH
 	depends on ALLOW_BOOT_FROM_DATAFLASH_CS1
 
 config	CONFIG_SPI_BOOT_CS2
 	bool	"Boot from chip select 2"
-	depends on CONFIG_DATAFLASH
 	depends on ALLOW_BOOT_FROM_DATAFLASH_CS2
 
 config	CONFIG_SPI_BOOT_CS3
 	bool	"Boot from chip select 3"
-	depends on CONFIG_DATAFLASH
 	depends on ALLOW_BOOT_FROM_DATAFLASH_CS3
 
 endchoice
diff --git a/driver/Config.in.memory b/driver/Config.in.memory
index 130ca76..ac14c1e 100644
--- a/driver/Config.in.memory
+++ b/driver/Config.in.memory
@@ -155,9 +155,13 @@ config CONFIG_LONG_FILENAME
 
 endmenu
 
-source "driver/Config.in.dataflash"
+if CONFIG_DATAFLASH
+	source "driver/Config.in.dataflash"
+endif
 
-source "driver/Config.in.nandflash"
+if CONFIG_NANDFLASH
+	source "driver/Config.in.nandflash"
+endif
 
 config CONFIG_BOOTSTRAP_MAXSIZE
 	string
diff --git a/driver/Config.in.nandflash b/driver/Config.in.nandflash
index 0332fb1..b0de70c 100644
--- a/driver/Config.in.nandflash
+++ b/driver/Config.in.nandflash
@@ -1,5 +1,4 @@
 menu  "NAND flash configuration"
-	depends on CONFIG_NANDFLASH
 
 choice
 	prompt "ECC Configuration"
@@ -81,7 +80,6 @@ config CONFIG_NANDFLASH_SMALL_BLOCKS
 config CONFIG_NANDFLASH_RECOVERY
 	bool "Support NAND flash recovery by pressing a button"
 	default y
-	depends on CONFIG_NANDFLASH
 	depends on ALLOW_NANDFLASH_RECOVERY
 	help
 	  Let bootstrap disassemble the first 7 double words
-- 
1.9.1

